src/components/StarRating/StarRating.ts

```diff
@@ -0,0 +1,63 @@
+import { $$, Utils, Component, Initialization, ComponentOptions, IComponentBindings } from 'coveo-search-ui';
+import { star as starIcon } from '../../utils/icons';
+
+export interface IStarRatingOptions {
+    /** Specifies the rating to be displayed as stars. */
+    rating: number;
+    /** Specifies the number to be displayed indicating the number of reviews. */
+    numberOfRatings?: number;
+}
+/**
+ * The `StarRating` component renders a 5-star rating widget.
+ *
+ * This component is a result template component (see [Result Templates](https://developers.coveo.com/x/aIGfAQ)).
+ */
+export class StarRating extends Component {
+    static ID = 'StarRating';
+
+    static options: IStarRatingOptions = {
+        rating: ComponentOptions.buildNumberOption({
+            defaultValue: 0
+        }),
+        numberOfRatings: ComponentOptions.buildNumberOption({
+            defaultValue: 0
+        })
+    };
+
+    /**
+     * Creates a new `StarRating` component.
+     * @param element The HTMLElement on which to instantiate the component.
+     * @param options The options for the `StarRating` component.
+     * @param bindings The bindings that the component requires to function normally. If not set, these will be
+     * automatically resolved (with a slower execution time).
+     */
+    constructor(public element: HTMLElement, public options: IStarRatingOptions, public bindings?: IComponentBindings) {
```

Comment:
you should probably get the "result" bindings and all here (check the ViewedByCustomer constructor for reference). This is also linked to my previous comment about passing "fields" instead of direct value :)

Reply:
I did it this way to give more flexibility to the user. Similarly to how the other ResultRating was unusable in our context, tying it to a result field might make it inadequate for another use case no?

Reply:
What do you envision as another use case? Other than that being bound to a field?

Reply:
Maybe I'm not understanding what you're saying. Being bound to a field means it would be bound to, let's say, @sfccrzproductratingc ?

Reply:
Yes, you don't want to have to write

<div class="CoveoStarRating" data-rating="4" data-number-of-rating="10">

in your result templates right? ;)

Reply:
I see I thought as it is putting `data-rating="@sfccrzproductratingc"` would work. I was under the impression the binding alternative would tie it to that explicitly named field. So the other use cases I was thinking of was in case anyone wanted to use this component in a non-ccrz context